post-processor/docker-push/post-processor.go

```diff
@@ -69,8 +70,27 @@ func (p *PostProcessor) PostProcess(ctx context.Context, ui packersdk.Ui, artifa
 
 	driver := p.Driver
 	if driver == nil {
+		ui.Message("Creating temporary Docker configuration directory")
+		tmpCfgDir, err := os.MkdirTemp("", "packer")
+		if err != nil {
+			return nil, false, false, fmt.Errorf(
+				"Error creating temporary Docker configuration directory: %s", err)
+		}
+
+		defer func() {
+			ui.Message("Removing temporary Docker configuration directory")
+			if err := os.RemoveAll(tmpCfgDir); err != nil {
+				ui.Error(
+					fmt.Sprintf("Error removing temporary Docker configuration directory: %s", err))
+			}
+		}()
+
 		// If no driver is set, then we use the real driver
-		driver = &docker.DockerDriver{Ctx: &p.config.ctx, Ui: ui}
+		driver = &docker.DockerDriver{
+			Ctx:       &p.config.ctx,
+			Ui:        ui,
+			ConfigDir: tmpCfgDir,
```

Comment:
Should we need to check if the environment variable DOCKER_CONFIG is set? 

The command line option will override any value defined in the environment variable if it is set do we want to create a new temp directory within DOCKER_CONFIG?

Reply:
Good point. I updated the logic to check if `DOCKER_CONFIG` is set. If it's not, it'll create the temporary directory.